src/mqtt_transport.rs

```diff
@@ -218,35 +205,19 @@ impl MqttTransport {
 
         let (read_socket, write_socket) = tokio::io::split(socket);
 
-        let mut mqtt_transport = Self {
+        let mqtt_transport = Self {
             token_source: Box::new(Arc::new(token_source)),
             write_socket: Arc::new(Mutex::new(write_socket)),
             read_socket: Arc::new(Mutex::new(read_socket)),
             d2c_topic: TopicName::new(cloud_bound_messages_topic(&device_id)).unwrap(),
             device_id: device_id.to_string(),
             #[cfg(feature = "c2d-messages")]
             rx_topic_prefix: device_bound_messages_topic_prefix(&device_id),
-            ping_join_handle: None,
             // rx_loop_handle: None,
         };
 
-        mqtt_transport.ping_join_handle = Some(Arc::new(mqtt_transport.ping_on_secs_interval(8)));
-
         Ok(mqtt_transport)
     }
-
-    ///
-    fn ping_on_secs_interval(&self, ping_interval: u8) -> JoinHandle<()> {
```

Comment:
Who's pinging now then?

Reply:
The user, manually.

Reply:
There is a `ping` method on `IoTHubClient`, so they'd just call that one.

Reply:
OK, so with "automatic-ping" on, we ask the library to set up a pinging task for us - the difference now is that we will receive all the ping reponses in our client right?